resourcemanager/identity/user_assigned_map.go

```diff
@@ -92,3 +92,56 @@ func FlattenUserAssignedMap(input *UserAssignedMap) (*[]interface{}, error) {
 		},
 	}, nil
 }
+
+// ExpandUserAssignedMapFromModel expands the typed schema input into a UserAssignedMap struct
+func ExpandUserAssignedMapFromModel(input []ModelUserAssigned) (*UserAssignedMap, error) {
+	if len(input) == 0 {
+		return &UserAssignedMap{
+			Type:        TypeNone,
+			IdentityIds: map[string]UserAssignedIdentityDetails{},
```

Comment:
pedantic, but we should probably remove this to be `nil` rather than an empty map (even if they're semantically the same):

```suggestion
```